content/guides/code-splitting.md

```diff
@@ -3,33 +3,43 @@ title: Code Splitting
 sort: 2
 contributors:
   - pksjce
+  - pastelsky
 ---
 
-Code splitting is the most compelling feature for  `webpack` usage. You can split your code into various bundles and load them on demand with `webpack`. It allows to tweak an application for these optimisations using the configuration.
-You can load them at a later time in your code or in a specific route only or on an event from the user even.
+Code splitting is one of the most compelling feature of webpack. It allows you to split your code into various bundles which you can then load on demand — like when a user navigates to a matching route, or on an event from the user. This allows for smaller bundles, and allows you to control resource load prioritization, which if used correctly, can have a major impact on your application load time.
 
-There are mainly two kind of code-splits that need to be accomplished with `webpack`
+There are mainly two kinds of code splitting that can be accomplished with webpack:
 
-## On demand code-splitting
+## Resource splitting for caching and parallel loads
 
-`webpack` can help us split our code into logical pieces or chunks as per our application routes or as per predicted user behaviour. This means that we can load non-essential assets when the user performs an action like route change and demands for it.
+### Vendor code splitting
 
-### Code splitting with `require.ensure()`
+A typical application can depend on many third party libraries for framework/functionality needs. Unlike the application code, code present in these libraries do not change very often. 
 
-`require.ensure()` is the CommonJS way of including assets asynchronously. By adding `require.ensure([<fileurl>])`, we can define a split point in the code. webpack can then create a separate bundle of all the code inside this split point.
-Learn [how to split code](/guides/code-splitting-require) using `require.ensure()`.
+If we keep code from these libraries onto its own bundle, separate from the application code, we can leverage browser's caching mechanism to cache these files for longer durations on the end user's machine. 
 
-?> Document `System.import()`
+For this to work, the `hash` portion in the vendor filename must remain constant, regardless of application code changes. Learn [how to split vendor/library](/guides/code-splitting-libraries) code using the CommonsChunkPlugin.
 
-## Resource splitting for cacheing and parallel loads
+T> Setting far cache expiry headers on your hashed resources allows browser to cache them for a longer time, reducing the need for re-downloading these resources by end users.
```

Comment:
I think this is a different topic (long therm caching) and should not be included here. I'm afraid that users get confused if they are tackled with multiple topics at once.